runtime/compiler/env/j9method.cpp

```diff
@@ -2386,15 +2387,48 @@ TR_ResolvedJ9Method::TR_ResolvedJ9Method(TR_FrontEnd * fe, TR_ResolvedMethod * o
 
 void TR_ResolvedJ9Method::construct()
    {
-#define x(a, b, c) a, sizeof(b) - 1, b, (int16_t)strlen(c), c
-
    struct X
       {
       TR::RecognizedMethod _enum;
       char _nameLen;
       const char * _name;
       int16_t _sigLen;
       const char * _sig;
+
+      typedef int8_t javaVersion_t;
+
+      javaVersion_t _nativeMinVersion;
+      javaVersion_t _nativeMaxVersion;
+
+// AllPastJavaVer and AllFutureJavaVer are used as lower and upper bounds for
+// ranges of Java versions over which a recognized method is expected to be
+// native, where that range is (currently) unbounded.  An upper bound of
+// AllPastJaveVer is used if the method has never been expected to be native.
+#define AllPastJavaVer (std::numeric_limits<int8_t>::min())
+#define AllFutureJavaVer (std::numeric_limits<int8_t>::max())
```

Comment:
Andrew @andrewcraik, I just noticed the recent [OMR pull request #4645](https://github.com/eclipse/omr/pull/4645).  Will my uses of `min()` and `max()` here be problematic on platforms that use xlC?

Reply:
I expect so

Reply:
Andrew @andrewcraik and Liqun @liqunl, I’m trying to use these definitions instead:

```
#define AllPastJavaVer INT8_MIN
#define AllFutureJavaVer INT8_MAX
```

But when compiling with xlC on z/OS or with xlC targetting Linux on Power, `INT8_MIN` and `INT8_MAX` are not defined by default.  They are defined if `__STDC_LIMIT_MACROS`  is defined before `stdint.h` is included.

There’s at least [one place in DDR](https://github.com/eclipse/omr/blob/b3fe6aaf4db093aaf07c6327157e34dd2b90b008/ddr/include/ddr/config.hpp#L30-L38) where I’ve seen code that defines `__STDC_LIMIT_MACROS`  only if `J9ZOS390` is defined, so I had been thinking I might do the same here if either `J9ZOS390` or `LINUXPPC` is defined.  However, in a conversation off-line Gita @gita-omr expressed slight reservations about using a platform-specific `#define`.

I’ve tried building across platforms with the definition of `__STDC_LIMIT_MACROS` guarded by `#if defined(J9ZOS390) || defined(LINUXPPC)` and without that guard, and things seems to work as expected either way.

As this is a standard mechanism for ensuring those macros are defined, I'll go ahead with defining `__STDC_LIMIT_MACROS` unconditionally, unless either of you has concerns.

Reply:
Made this change in commit https://github.com/eclipse/openj9/pull/6598/commits/6bf70869e38c2ba4bff686de6b787e5ecb8477a7.  If you're OK with it, I will attempt to squash this new commit into commit https://github.com/eclipse/openj9/pull/6598/commits/ca96073a1846af1cdd0d31faee6178e96958e4d3